Marlin/src/feature/easythreed_ui.cpp

```diff
@@ -0,0 +1,242 @@
+
+    case KS_PRESS:
+      if (ELAPSED(ms, key_time + BTN_DEBOUNCE_MS))                  // Wait for debounce interval to expire
+        key_status = READ(BTN_PRINT) ? KS_IDLE : KS_PROCEED;        // Proceed if still pressed
+      break;
+
+    case KS_PROCEED:
+      if (!READ(BTN_PRINT)) break;                                  // Wait for the button to be released
+      key_status = KS_IDLE;                                         // Ready for the next press
+      if (PENDING(ms, key_time + 1200 - BTN_DEBOUNCE_MS)) {         // Register a press < 1.2 seconds
+        switch (print_key_flag) {
+          case PF_START:                                            // The "Print" button starts an SD card print
+            if (!printingIsActive()) {                              // Not already printing?
+              card.mount();                                         // Force SD card to mount - now!
+              if (!card.isMounted) {                                // Failed to mount?
+                blink_interval_ms = LED_OFF;                        // Turn off LED
+                print_flag = false;                                 // Clear previous print flag
+                return;                                             // Bail out
+              }
+              card.ls();                                            // List all files to serial output
+              const uint16_t filecnt = card.countFilesInWorkDir();  // Count printable files in cwd
+              if (filecnt == 0) return;                             // None are printable?
+              card.selectFileByIndex(filecnt);                      // Select the last file according to current sort options
+              card.openAndPrintFile(card.filename);                 // Start printing it
+              blink_interval_ms = LED_BLINK_2;                      // Blink the indicator LED at 1 second intervals
+              print_flag = true;                                    // Tell EasythreedUI printing should be active
+              print_key_flag = PF_PAUSE;                            // The "Print" button now pauses the print
+            }
+            break;
+          case PF_PAUSE:                                            // Pause printing (not currently firing)
+            blink_interval_ms = LED_ON;                             // Set indicator to steady ON
+            card.pauseSDPrint();                                    // Unceremoniously pause the SD print in progress
+            print_key_flag = PF_RESUME;                             // The "Print" button now resumes the print
+            break;
+          case PF_RESUME:                                           // Resume printing (not currently firing)
+            blink_interval_ms = LED_BLINK_2;                        // Blink the indicator LED at 1 second intervals
+            card.startOrResumeFilePrinting();                       // Unceremoniously resume the SD print in progress
```

Comment:
It is better to send `M24` because this only resumes SD card processing. It doesn't un-park, etc.